pkg/vgmanager/vgmanager_controller.go

```diff
@@ -413,3 +423,67 @@ func (r *VGReconciler) getMatchingDevicesForVG(volumeGroup *lvmv1alpha1.LVMVolum
 
 	return matchingDevices, delayedDevices, nil
 }
+
+func (r *VGReconciler) generateVolumeGroupNodeStatus(lvmdConfig *lvmdCMD.Config) (*lvmv1alpha1.LVMVolumeGroupNodeStatus, error) {
+
+	vgs, err := ListVolumeGroups(r.executor)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list volume groups. %v", err)
+	}
+
+	//lvmvgstatus := vgNodeStatus.Spec.LVMVGStatus
+	var statusarr []lvmv1alpha1.VGStatus
+
+	for _, vg := range vgs {
+		newStatus := &lvmv1alpha1.VGStatus{
+			Name:    vg.Name,
+			Reason:  "test",
+			Devices: vg.PVs,
+		}
+		statusarr = append(statusarr, *newStatus)
+	}
+
+	vgNodeStatus := &lvmv1alpha1.LVMVolumeGroupNodeStatus{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.NodeName,
+			Namespace: r.Namespace,
+		},
+		Spec: lvmv1alpha1.LVMVolumeGroupNodeStatusSpec{
+			LVMVGStatus: statusarr,
+		},
+	}
+
+	return vgNodeStatus, nil
+}
+
+func (r *VGReconciler) updateStatus(ctx context.Context, lvmdConfig *lvmdCMD.Config) error {
```

Comment:
- I don't see `lvmdConfig` is being used directly in this func or other funcs being called in this func
- If it's intentional we don't need this param at all?
- I was thinking this can be used to filter the vgs that lvmo is managing?

Reply:
ack, makes sense, we can use tags then if that's still an options, maybe in another pr.